### PR TITLE
Emisweb bulk 99

### PR DESCRIPTION
**Pull request by sekiskylink**
- Added code to format Student enrolments based on DHIS2 schema
- Added functionality to dry run enrolment and also commit the same.

### CODE REVIEW

File: src/hooks/bulkImport/postTrackedEntities.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { atom, useRecoilState } from "recoil";
+import {ApiResponse} from "../../types/bulkImport/Interfaces";
+
+const POST_TEI: any = {
+    resource: "tracker",
+    type: 'create',
+    data: ({ data }: any) => data,
+    params: ({params}: any) => params
+}
+
```
Comment (Eddy-Nhabinde) [inline src/hooks/bulkImport/postTrackedEntities.ts:12]: type missing

File: src/hooks/bulkImport/postTrackedEntities.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export const teiRefetch = atom({
+    key: "refetch-tei",
+    default: false
+})
```
Comment (ClaydLena) [inline src/hooks/bulkImport/postTrackedEntities.ts:16]: Schemas should be on schemas folder

File: src/components/bulkImport/SummaryDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    const processedRecords: ProcessingRecords = useRecoilValue<ProcessingRecords>(ProcessingRecordsState)
+    const [activeTab, setActiveTab] = useState("invalids")
+    // const [page, setPage] = useState<number>(1)
+    // const pageSize = 5
+    const [pagination, setPagination] = useState<PaginationState>({
+        new: {page: 1, pageSize: 5},
+        updates: {page: 1, pageSize: 5},
+        conflicts: {page: 1, pageSize: 5},
+        invalids: {page: 1, pageSize: 5}
+    });
```
Comment (ClaydLena) [inline src/components/bulkImport/SummaryDetails.tsx:28]: This object should be on constants folder

**Review by CelsoDeCarvalho — APPROVED** [review]

File: src/components/bulkImport/SummaryDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                        <DataTableCell>{""}</DataTableCell>
+                    </DataTableRow>
+                ))}
+                {(displayData?.length === 0) &&
+                    <DataTableCell>{"No Data"}</DataTableCell>
+                }
+            </DataTableBody>
+        </DataTable>
+        <br/>
+        {total > 0 && <Pagination
```
Comment (AlziraXerinda) [inline src/components/bulkImport/SummaryDetails.tsx:120]: We can use the component already created for pagination. 
[Pagination.tsx](https://github.com/Saudigitus/dhis2-emis-enrollment/blob/develop/src/components/table/components/pagination/Pagination.tsx)

**Comment by AlziraXerinda** [comment]
Post a suggestive topic for the pull request.

**Review by AlziraXerinda — APPROVED** [review]

File: src/components/bulkImport/BulkEnrollment.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+                return
+            }
+            setIsValidTemplate(true)
+            const headers: string[] = rawData[1] as string[]
+            const dataWithHeaders: Array<Record<string, any>> = generateData(headers, rawData.slice(2))
+            const fieldMapping = fieldsMap(programConfig, enrollmentStages)
+            // console.log("<<<<<<", fieldMapping)
+            const [invalidRecords, validRecords, newRecords, recordsToUpdate] = await processData(
+                dataWithHeaders, fieldMapping, programConfig, engine)
+            console.log("INVALID RECORDS:", invalidRecords)
```
Comment (EdsonNhancale) [inline src/components/bulkImport/BulkEnrollment.tsx:119]: [Improvement] Try to move this code to some custom hook @sekiskylink

**Pull request merged**